jack-core/src/com/rapleaf/jack/transaction/ITransactor.java

```diff
@@ -23,4 +29,14 @@
     Impl get();
   }
 
+  interface RetryPolicy {
+
+    boolean shouldRetry();
+
+    void updateOnFailure();
```

Comment:
Agree completely on the first part. I had originally named it onFailure and onSuccess, I am not sure why I changed that. :-/ I realize that on{EventName} is common for handling events (My Java is still a bit rusty after having worked on C all this while).

Probably a good idea to allow `onFailure` to handle the exception -- which makes me think that the `shouldRetry` method is redundant since `onFailure` could simply throw/propagate the exception to exit the loop. Thoughts?

Reply:
Hmm, that's also another way of looking at it. So the usage would be like the following, correct? 

- try to execute a query
- catch an exception
- in the catch block, call `onFailure(e)`
- if the number of retries is below a certain number, then you increment the number and execute what is now in the `execute` method
- If the number of retries is above the threshold, then you rethrow the exception allowing the method to fail normally

Maybe this whole thing could just be `onFailure`? What do you think?

Reply:
That was the original thought; However, the reason I have `onSuccess` and `execute` is to allow the policy to sleep (or do some other blocking/long running task) after returning the connection to the connection pool, which is located in the `finally` block. Since the `finally` block executes no matter what, there must be some way of knowing when not to sleep -- either the number of retries have been exhausted or the query was successful.